src/core/public/overlays/overlay_service.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { FlyoutService } from './flyout';
+
+import { I18nSetup } from '../i18n';
+
+interface Deps {
+  i18n: I18nSetup;
+}
+
+export class OverlayService {
+  private flyoutService: FlyoutService = new FlyoutService();
+
+  public setup({ i18n }: Deps) {
```

Comment:
Not sure if we need a `stop()` interface to do cleanup?

Reply:
If you move the `i18n` to be passed into the constructor of the FlyoutService and move instantiation to the `setup`. Then perhaps you want a stop method to possibly close the flyout and delete the service object?